### ISSUE TITLE
Support running CapX in Docker

### ISSUE THREAD

**Issue by ianhinder** [UoMResearchIT/CapX#153]
It would be convenient to be able to run CapX in Docker for development without needing to install the build tools in your local environment.

**Comment by ianhinder** [comment]
A work-in-progress implementation in Docker is in [153-support-running-capx-in-docker](https://github.com/UoMResearchIT/CapX/tree/153-support-running-capx-in-docker).  It doesn't yet work due to #151 and #152.

**Comment by aharwood2** [comment]
OK, so I never imagined that someone would try to develop a .NET application without using an IDE -- if you were to install VS it just setups up everything out the box and you just hit play. Deployment is self-contained so it took more time to setup a dockerfile than it does to just one-click deploy it. That said, if you want to dockerise it, that's fine but I just wanted to let you know that it isn't the most efficient to develop .NET apps, even on Mac.

**Comment by aharwood2** [comment]
@ianhinder Looks like #152 is now closed and #151 could be closed so are you ready to raise a PR for this? I would probably advise that the README updated to include  a prompt to ensure EF core tools are installed and the migrations applied (see your comment on #152) -- perhaps as part of the dockerfile so the DB is ready to go once the container starts?

**Comment by aharwood2** [comment]
Bumped to v1.9 as still need to have a think about running the migrations when a new DB is created -- I guess could be done as part of building the image so an initialised DB file is embedded in the image? I'll leave it with you to think about.

**Comment by aharwood2** [comment]
Not sure where we left this now, was it just deciding how to run the migrations? You can do this with a call to the ef core tool probably as part of the image build?

**Comment by ianhinder** [comment]
Yeah I was able to get it up and running, I think.  I'll take another look and try to put this one to bed in the next week or so.

**Comment by aharwood2** [comment]
@ianhinder I've circled back round to this as there might be a few more people chipping in to help add features soon and I thought it best we make it as easy as possible for cross-platform development. In my limited understanding of docker, I've got a solution that works and updated the readme instructions accordingly. It might not be the preferred method but it is functional at least. Would appreciate your review of it so will raise a PR and add you as a reviewer if that is OK?
